tests/integration/dependencies_test.go

```diff
@@ -0,0 +1,264 @@
+			},
+			unexpectedEvents: []events.ID{events.FailedAttach},
+			expectedEvents:   []events.ID{events.SubmitOnce},
+		},
+	}
+
+	logOutChan, restoreLogger := testutils.SetTestLogger(logger.DebugLevel)
+
+	for _, testCaseInst := range testCases {
+		t.Run(
+			testCaseInst.name, func(t *testing.T) {
+				// prepare tracee config
+				testConfig := config.Config{
+					Capabilities: &config.CapabilitiesConfig{
+						BypassCaps: true,
+					},
+				}
+				testConfig.Policies = buildPoliciesFromEvents(testCaseInst.events)
+				policy.Snapshots().Store(testConfig.Policies)
+
+				ctx, cancel := context.WithCancel(context.Background())
+				done := make(chan struct{})
+				logsResultChan := testLogs(t, testCaseInst.expectedLogs, logOutChan, done)
+
+				// start tracee
+				trc, err := startTracee(ctx, t, testConfig, nil, nil)
+				if err != nil {
+					cancel()
+					t.Fatal(err)
+				}
+				t.Logf("  --- started tracee ---")
+
+				stream := trc.SubscribeAll()
+				defer trc.Unsubscribe(stream)
+				eventsResultChan := testEvents(
+					t,
+					testCaseInst.expectedEvents,
+					testCaseInst.unexpectedEvents,
+					stream.ReceiveEvents(),
+					done,
+				)
+
+				err = waitForTraceeStart(trc)
+				if err != nil {
+					cancel()
+					t.Fatal(err)
+				}
+
+				// Give the events a chance to be received and pass the pipeline
+				time.Sleep(3 * time.Second)
+
+				cancel()
+				errStop := waitForTraceeStop(trc)
+				if errStop != nil {
+					t.Log(errStop)
+				} else {
+					t.Logf("  --- stopped tracee ---")
+				}
+				close(done)
+
+				assert.True(t, <-logsResultChan)
+				assert.True(t, <-eventsResultChan)
+			},
+		)
+	}
+	restoreLogger()
+}
+
+// testLogs searches for given logs and test when input channel closes if all
+// logs were received.
+// It also returns a channel with the result of the test - whether all logs were found.
+func testLogs(
+	t *testing.T,
+	logsToSearch []string,
+	logsChan <-chan []byte,
+	done <-chan struct{},
+) <-chan bool {
+	testResults := make(map[string]bool, len(logsToSearch))
+	for _, log := range logsToSearch {
+		testResults[log] = false
+	}
+
+	outChan := make(chan bool)
+
+	go func() {
+	Loop:
+		for {
+			select {
+			case receivedLog, ok := <-logsChan:
+				if !ok {
+					break Loop
+				}
+				for _, logToSearch := range logsToSearch {
+					if strings.Contains(string(receivedLog), logToSearch) {
+						testResults[logToSearch] = true
+					}
+				}
+			case <-done:
+				break Loop
+			}
+		}
+
+		for logToSearch, found := range testResults {
+			assert.True(t, found, logToSearch)
+			if !found {
+				outChan <- false
+				close(outChan)
+				return
+			}
+		}
+		outChan <- true
+		close(outChan)
+	}()
+	return outChan
+}
+
+// testEvents searches for the given events in the events channel, and when the
+// run is completed test whether the given events have been found.
+// It gives the option for searching for event that should and shouldn't be
+// found to enable precise tests.
+func testEvents(
```

Comment:
I believe @geyslan made similar functions in his filter tests. Impossible to reuse them?

Reply:
I will look into it